aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -250,25 +251,28 @@ public SdkHttpFullRequest sign(
     } else {
       Map<String, String> responseHeaders = Maps.newHashMap();
       Consumer<Map<String, String>> responseHeadersConsumer = responseHeaders::putAll;
-      S3SignResponse s3SignResponse =
-          httpClient()
-              .withAuthSession(authSession())
-              .post(
-                  endpoint(),
-                  remoteSigningRequest,
-                  S3SignResponse.class,
-                  Map.of(),
-                  ErrorHandlers.defaultErrorHandler(),
-                  responseHeadersConsumer);
-
-      signedComponent =
-          ImmutableSignedComponent.builder()
-              .headers(s3SignResponse.headers())
-              .signedURI(s3SignResponse.uri())
-              .build();
-
-      if (canBeCached(responseHeaders)) {
-        SIGNED_COMPONENT_CACHE.put(cacheKey, signedComponent);
+      try (RESTClient restClient = httpClient().withAuthSession(authSession())) {
```

Comment:
this effectively closes nothing in particular (unless I'm missing something)?

Reply:
Yes, this is purely for correctness.

Reply:
given that we're not closing anything I would not wrap this in a try-with-resources here, because I had to go over this code a few times to make sure we're not causing a performance degradation, since this part will be called quite often during signing. Other readers of the code might have the same thought and arrive at the same conclusion after studying the code and that's why I think it's better to not use try-with-resources here (and most likely in all of the places in the `RESTSessionCatalog`

Reply:
OK removing the try with resources blocks